### ISSUE TITLE
language-c-0.8 fails to parse fa.h / stdio.h

### ISSUE THREAD

**Issue by e1528532** [visq/language-c#55]
Hello,

I wrote a small binding to the library libfa (http://augeas.net/libfa/) with c2hs.

However, i noticed that when c2hs is compiled with the newer language-c-0.8 instead of language-c-0.7.2 it fails to parse the header file of libfa. With the older version it works. This is also the reason why i post the issue here instead of the c2hs repository. In case you think this is unrelated to language-c feel free to redirect me.
You'll see the exact issue at the end of the build log, it fails to parse the standard stdio.h header file that is included by libfa's header file. My version of stdio.h is the one bundled with Apple LLVM version 9.1.0 (clang-902.0.39.2) as i am using macOS 10.13.4. I have additionally attached the lines of stdio.h that cause the failure.

Please find attached the build log, i tidied it up a bit to only include the relevant information:

```
Failed to install libfa-1.0.8.23
Build log:
Using Parsec parser
Configuring libfa-1.0.8.23...
Dependency base ==4.10.1.0: using base-4.10.1.0
Source component graph: component lib
Configured component graph:
    component libfa-1.0.8.23-ZfhKuK3orHDHB6gQt3Gon
        include base-4.10.1.0
Linked component graph:
    unit libfa-1.0.8.23-ZfhKuK3orHDHB6gQt3Gon
        include base-4.10.1.0
        FiniteAutomata=libfa-1.0.8.23-ZfhKuK3orHDHB6gQt3Gon:FiniteAutomata
Ready component graph:
    definite libfa-1.0.8.23-ZfhKuK3orHDHB6gQt3Gon depends base-4.10.1.0
Using Cabal-2.2.0.0 compiled by ghc-8.4
Using compiler: ghc-8.2.2
Using alex version 3.2.4 found on system at: /Users/travis/.cabal/bin/alex
Using ar found on system at: /usr/bin/ar
Using c2hs version 0.28.5 found on system at: /Users/travis/.cabal/bin/c2hs
No cpphs found
No doctest found
Using gcc version 4.2.1 found on system at: /usr/bin/clang
Using ghc version 8.2.2 found on system at: /usr/local/bin/ghc
Using ghc-pkg version 8.2.2 found on system at: /usr/local/bin/ghc-pkg
No ghcjs found
No ghcjs-pkg found
No greencard found
Using haddock version 2.18.1 found on system at: /usr/local/bin/haddock
Using happy version 1.19.9 found on system at: /Users/travis/.cabal/bin/happy
Using haskell-suite found on system at: haskell-suite-dummy-location
Using haskell-suite-pkg found on system at: haskell-suite-pkg-dummy-location
No hmake found
Using hpc version 0.67 found on system at: /usr/local/bin/hpc
Using hsc2hs version 0.68.2 found on system at: /usr/local/bin/hsc2hs
No hscolour found
No jhc found
Using ld found on system at: /usr/bin/ld
No lhc found
No lhc-pkg found
Using pkg-config version 0.29.2 found on system at: /usr/local/bin/pkg-config
Using runghc version 8.2.2 found on system at: /usr/local/bin/runghc
Using strip found on system at: /usr/bin/strip
Using tar found on system at: /usr/bin/tar
No uhc found
Component build order: library
Preprocessing library for libfa-1.0.8.23..

/Users/travis/.cabal/bin/c2hs '--cpp=/usr/bin/clang' '--cppopts=-E' '--cppopts=-D__GLASGOW_HASKELL__=802' '--cppopts=-Ddarwin_BUILD_OS=1' '--cppopts=-Dx86_64_BUILD_ARCH=1' '--cppopts=-Ddarwin_HOST_OS=1' '--cppopts=-Dx86_64_HOST_ARCH=1' '--cppopts=-I/usr/local/include' '--cppopts=-includedist/dist-sandbox-c7261b7a/build/autogen/cabal_macros.h' '--include=dist/dist-sandbox-c7261b7a/build' '--cppopts=-I/usr/local/Cellar/ghc@8.2/8.2.2/lib/ghc-8.2.2/base-4.10.1.0/include' '--cppopts=-I/usr/local/Cellar/ghc@8.2/8.2.2/libexec/integer-gmp/include' '--cppopts=-I/usr/local/Cellar/ghc@8.2/8.2.2/lib/ghc-8.2.2/integer-gmp-1.0.1.0/include' '--cppopts=-I/usr/local/Cellar/ghc@8.2/8.2.2/lib/ghc-8.2.2/include' '--output-dir=dist/dist-sandbox-c7261b7a/build' '--output=FiniteAutomata.hs' ./FiniteAutomata.chs

c2hs: C header contains errors:
/usr/include/_stdio.h:137: (column 19) [ERROR]  >>> Syntax error !
  The symbol `_close' does not fit here.
```

The cabal file looks like this:

```
name:                libfa
version:             0.1.0.0
build-type:          Simple
cabal-version:       >=1.24

library
  exposed-modules:   FiniteAutomata
  build-depends:     base >= 4.9 && < 4.12
  includes:          fa.h
  include-dirs:      "@LIBFA_INCLUDE_DIRS@"
  extra-lib-dirs:    "@LIBFA_LIBRARY_DIRS@"
  extra-libraries:   fa
  default-language:  Haskell2010
  other-extensions:  ForeignFunctionInterface
  build-tools:       c2hs
```

And a small excerpt of _stdio.h, see line 137:

```
   126  typedef struct __sFILE {
   127    unsigned char *_p;  /* current position in (some) buffer */
   128    int _r;   /* read space left for getc() */
   129    int _w;   /* write space left for putc() */
   130    short _flags;   /* flags, below; this FILE is free if 0 */
   131    short _file;    /* fileno, if Unix descriptor, else -1 */
   132    struct  __sbuf _bf; /* the buffer (at least 1 byte, if !NULL) */
   133    int _lbfsize; /* 0 or -_bf._size, for inline putc */
   134  
   135    /* operations */
   136    void  *_cookie; /* cookie passed to io functions */
   137    int (* _Nullable _close)(void *);
   138    int (* _Nullable _read) (void *, char *, int);
   139    fpos_t  (* _Nullable _seek) (void *, fpos_t, int);
   140    int (* _Nullable _write)(void *, const char *, int);
   141  
   142    /* separate buffer for long sequences of ungetc() */
   143    struct  __sbuf _ub; /* ungetc buffer */
   144    struct __sFILEX *_extra; /* additions to FILE to not break ABI */
   145    int _ur;    /* saved _r when _r is counting ungetc data */
   146  
   147    /* tricks to meet minimum requirements even when malloc() fails */
   148    unsigned char _ubuf[3]; /* guarantee an ungetc() buffer */
   149    unsigned char _nbuf[1]; /* guarantee a getc() buffer */
   150  
   151    /* separate buffer for fgetln() when line crosses buffer boundary */
   152    struct  __sbuf _lb; /* buffer for fgetln() */
   153  
   154    /* Unix stdio files get aligned to block boundaries on fseek() */
   155    int _blksize; /* stat.st_blksize (may be != _bf._size) */
   156    fpos_t  _offset;  /* current lseek offset (see WARNING) */
   157  } FILE;
```

**Comment by visq** [comment]
@e1528532 thank you for the detailled report.
This is indeed a regression in 0.8; the master for the list of tokens was missing _Nullable and _Noreturn.
I will fix this immediately in 0.8.1, and add a regression test ensuring all tokens are recognized.

**Comment by visq** [comment]
Released 0.8.1 that fixes the regression
